draft-irtf-cfrg-hash-to-curve.md

```diff
@@ -831,134 +831,207 @@ Instantiating one of these protocols with an elliptic curve group motivates
 the term "hashing to the curve", i.e., encoding bitstrings to points on an
 elliptic curve.
 
-The hash2curve(alpha) function can be easily constructed by
-using the output of a cryptographically secure hash function H as the input of
-an encoding function.
-On the one hand, hash2curve is difficult to invert since it is computationally
-intractable to produce an input alpha that maps to hash2curve(alpha) due to H
-is pre-image resistant.
+One can easily construct a hash to curve function h(x) by using the output of a
+cryptographically secure hash function H as the input of an encoding function.
+On the one hand, h(x) is difficult to invert since it is computationally
+intractable to produce an input x that maps to h(x) due to H is pre-image
+resistant.
 On the other hand, the uniformity property is not met as the output of an
 encoding is distinguishable from a random distribution. Hence, using
-hash2curve(alpha) is not sufficient to get uniformity, however it can be used
-as a building block for obtaining a random oracle as is described in {{rom}}.
+h(x) is not sufficient to get uniformity and cannot be used as a random oracle.
+
+Brier et al. {{BCIMRT10}} describe two generic constructions that give a
+hash function approximating a random oracle. Farashahi et al. {{FFSTV13}}
+and Tibouchi and Kim {{TK17}} refine this analysis. In particular, Farashahi
+et al. show that summing two independent evaluations of an injective encoding
+suffices to approximate a random oracle to an elliptic curve. This construction
+is given in {{roadmap}}.
+
+# Roadmap {#roadmap}
+
+A general framework for mapping bit strings into points of an elliptic curve is
+presented. The construction of these mappings relies on three basic functions.
+
+-   The function hash2base, {0, 1}^* -> F, maps arbitrary-length bit strings
+    to elements of a finite field; its implementation is defined in
+    {{hashtobase}}.
+
+-   The function map2curve, F -> E, calculates a point on the elliptic curve E
+    from an element of a finite field. {{encodings}} describes mappings for a
+    range of curve families.
+
+-   The function clear\_cofactor, E -> G, maps any point on the curve E to a
+    subgroup G of the curve. {{cofactor-clearing}} describes methods to perform
+    this operation.
+
+Two top-level functions taking as input a bit string and returning points on
+the curve are introduced. Although these functions have the same interface, the
+distribution of the points they produce are different. The first function is
+called injective encoding and its probability distribution of the points can be
+easily distinguished from a uniform distribution of points on the curve. The
+second function behaves as a hash function since its output is
```

Comment:
Here I think "hash function" is confusing, because not all hash functions can be modeled as random oracles. Certainly collision resistance doesn't imply usability as a PRF, for example.

Maybe "behaves as a random oracle, since its output" instead?